contracts/cw721-sellable/src/lib.rs

```diff
@@ -107,7 +111,12 @@ mod entry {
             contract
                 .mint(heap_deps.branch(), env.clone(), info.clone(), mint_msg)
                 .unwrap();
+            tokens_to_list.insert(n.to_string(), msg.contract_metadata.initial_price);
```

Comment:
Wouldn't it make sense to try listing these as we find them? We are basically iterating over them twice, once to put them into a map, and again by iterating over that map.

Reply:
@Peartes A good point by @ash-burnt. When we iterate through to mint, let's just add the metadata to the mint message that includes the list price. That way we can do it in one fell swoop.